pkg/buildchange/stack_change.go

```diff
@@ -0,0 +1,50 @@
+package buildchange
+
+import (
+	"strings"
+
+	"github.com/google/go-containerregistry/pkg/name"
+	"github.com/pkg/errors"
+
+	"github.com/pivotal/kpack/pkg/apis/build/v1alpha1"
+)
+
+func NewStackChange(oldRunImageRefStr, newRunImageRefStr string) Change {
+	var change stackChange
+	var errStrs []string
+
+	oldRunImageRef, err := name.ParseReference(oldRunImageRefStr)
+	if err != nil {
+		errStrs = append(errStrs, err.Error())
+	} else {
+		change.oldRunImageId = oldRunImageRef.Identifier()
+	}
+
+	newRunImageRef, err := name.ParseReference(newRunImageRefStr)
+	if err != nil {
+		errStrs = append(errStrs, err.Error())
+	} else {
+		change.newRunImageId = newRunImageRef.Identifier()
+	}
+
+	if len(errStrs) > 0 {
+		change.err = errors.New(strings.Join(errStrs, "; "))
+	}
+	return change
+}
+
+type stackChange struct {
+	oldRunImageId string
```

Comment:
Could we call these digests instead of id?

Reply:
It was called `oldRunImageId` because it holds the image identifier (which happens to be a digest) 
```
change.oldRunImageId = oldRunImageRef.Identifier()
```

Reply:
marking as resolved. made the change anyway with commit https://github.com/pivotal/kpack/commit/6bcb2c1a346ae2149b5fd1d2681cf759f93ae3a7